help_to_heat/frontdoor/views.py

```diff
@@ -358,7 +360,107 @@ def save_data(self, request, session_id, page_name, *args, **kwargs):
 @register_page("address")
 class AddressView(PageView):
     def handle_post(self, request, session_id, page_name, data, is_change_page):
-        return redirect("frontdoor:page", session_id=session_id, page_name="address-select")
+        building_name_or_number = data["building_name_or_number"]
+        postcode = data["postcode"]
+        country = interface.api.session.get_answer(session_id, "country")["country"]
+        try:
+            if country == "Scotland":
+                return redirect("frontdoor:page", session_id=session_id, page_name="address-select")
+            else:
+                address_and_rrn_details = interface.api.epc.get_address_and_epc_rrn(building_name_or_number, postcode)
+                interface.api.session.save_answer(
+                    session_id, page_name, {"address_and_rrn_details": address_and_rrn_details}
+                )
+                return redirect("frontdoor:page", session_id=session_id, page_name="epc-select")
+        except Exception as e:  # noqa: B902
+            logger.exception(f"An error occurred: {e}")
+            interface.api.session.save_answer(
```

Comment:
@jamiehumphries we effectively reset the test data as, if someone inputs address details for which the EPC API does find results, and then goes back and changes their address and no results are found, or they input it manually, old EPC details for the first address will still be in the session. I've extracted a top-level method for dealing with the resetting, but I do wonder if we should also reset details in the Scotland branch of the if statement. It's a very unlikely edge case, but just in case a user changes from an address in England (where there was a successful EPC look-up) to one in Scotland, where we don't perform EPC API lookups.

Reply:
Ah, I see. Thanks!

What you say makes sense. In fact, perhaps we should just always reset the data at the top of this `handle_post`? Regardless of any other factors, we are always going to want to discard the old data when handling a new post (I think).

Reply:
(I was going to react with a + but GitHub doesn't allow that.)

I agree that always resetting at the top of `handle_post` should be fine (with the benefit of slightly more familiarity with this code) 🙂